powertools-metrics/src/main/java/software/amazon/lambda/powertools/metrics/MetricsUtils.java

```diff
@@ -81,16 +81,10 @@ public static void withSingleMetric(final String name,
                                         final double value,
                                         final Unit unit,
                                         final Consumer<MetricsLogger> logger) {
-        MetricsLogger metricsLogger = logger();
-
-        try {
-            metricsLogger.setNamespace(defaultNameSpace());
+        withMetric(metricsLogger -> {
             metricsLogger.putMetric(name, value, unit);
```

Comment:
`putMetric` is no longer within the `try`/`finally` block. As far as I can see it doesn't throw any runtime exceptions - did you dive into this to make sure?

Reply:
The way I thought about this, is that whatever was happening in the `try` block, remains happening in a `try` block. It's just that it moved to the new `withMetric` method.

Reply:
You're right - I misread the lambda. Cool!